packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -0,0 +1,1466 @@
+
+        emit KghUndelegationFinalized(validator, msg.sender, assets);
+        return assets;
+    }
+
+    /**
+     * @notice Withdraw the reward of the validator.
+     */
+    function withdrawValidatorReward() external {
+        uint128 reward = vault[msg.sender].reward.validatorRewardKro;
+        require(reward > 0, "AssetManager: No reward to withdraw");
+
+        vault[msg.sender].reward.validatorRewardKro = 0;
+        bool success = IERC20(GOVERNANCE_TOKEN).transfer(msg.sender, reward);
+        require(success, "AssetManager: Transfer failed");
+    }
+
+    /**
+     * @notice Internal function to add base and boosted reward to the vaults of finalized output submitters.
+     */
+    function _distributeReward() internal returns (bool) {
+        // TODO - Query the index from L2OutputOracle
+        uint256 outputIndex = nextRewardOutputIndex;
+        uint128 rewardedNum = 0;
+        Types.CheckpointOutput memory output;
+
+        for (; rewardedNum < MAX_REWARD_RECIPIENTS; ) {
+            uint256 expiryTime = submittedOutputs[outputIndex];
+            if (block.timestamp >= expiryTime && expiryTime != 0) {
+                delete submittedOutputs[outputIndex];
+                output = L2_ORACLE.getL2Output(outputIndex);
+                _increaseBalanceWithReward(output.submitter);
+
+                uint128 challengeReward = pendingChallengeReward[outputIndex];
+                if (challengeReward > 0) {
+                    _modifyBalanceWithSlashing(vault[output.submitter], challengeReward, false);
+                    delete pendingChallengeReward[outputIndex];
+                }
+
+                unchecked {
+                    ++outputIndex;
+                    ++rewardedNum;
+                }
+            } else {
+                break;
+            }
+        }
+
+        if (rewardedNum > 0) {
+            unchecked {
+                nextRewardOutputIndex = outputIndex;
+            }
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @notice Slash KRO at the vault of the validator
+     *
+     * @param loser  Address of the loser at the challenge
+     * @param winner Address of the winner at the challenge
+     */
+    function slash(address loser, address winner, uint256 outputIndex) external onlyColosseum {
+        uint128 amountToSlash = _modifyBalanceWithSlashing(vault[loser], 0, true);
+        pendingChallengeReward[outputIndex] = amountToSlash;
+
+        _updateWeightTree(loser);
+
+        emit Slashed(loser, winner, amountToSlash);
+    }
+
+    /**
+     * @notice Add pending assets and shares when undelegating KRO.
+     *
+     * @param _vault Vault of the validator.
+     * @param assets The amount of assets to add as pending asset.
+     * @param shares The amount of shares to add as pending share.
+     */
+    function _addPendingKroShares(Vault storage _vault, uint128 assets, uint128 shares) internal {
```

Comment:
The reason why I named it `_vault` was to separate from the mapping `vault`. The naming convention is typically applied to the internal functions and state variables, not to the parameters.

But according to the convention, it would be better to add underscore prefix to the following internal state variables:
- `weightTree`
- `nextRewardOutputIndex`
- `vault`
- `submittedOutputs`
- `pendingChallengeReward`

and then it would be better to change `Vault storage _vault` to `Vault storage vault` in this file. How do you think about this?

Reply:
Oh I see it, that sounds good.